packages/actions/types.ts

```diff
@@ -0,0 +1,43 @@
+import type { Products } from '@hashicorp/platform-product-meta'
+
+interface CtaProps {
+  /**
+   * The text that appears inside the button.
+   */
+  title: string
+  /**
+   * Where the button links to when clicked.
+   */
+  url: string
+  /**
+   * Applies a styling to the button based on the desired hierarchy.
+   */
+  variant?: 'primary' | 'secondary' | 'tertiary-neutral'
+  /**
+   * A function that will be called when the button is clicked.
+   */
+  onClick?: () => void
+}
+
+type SingleCta = [CtaProps]
+type DuoCta = [CtaProps, CtaProps]
+
+export interface ActionsProps {
+  /**
+   * Display buttons inline or stacked by default.
+   */
+  layout?: 'inline' | 'stacked'
+  /**
+   * Render primary button with product color.
+   */
+  brand?: Products
+  /**
+   * Determines CTA button sizing.
+   */
+  size?: 'small' | 'medium'
+  /**
+   * Array of CTAs. Minimum of one, max of two.
+   * @see CtaProps
+   */
+  ctas: SingleCta | DuoCta
```

Comment:
This could be simplified I think:

```suggestion
type MaxTwoCta = readonly [CtaProps, CtaProps?]

export interface ActionsProps {
  /**
   * Display buttons inline or stacked by default.
   */
  layout?: 'inline' | 'stacked'
  /**
   * Render primary button with product color.
   */
  brand?: Products
  /**
   * Determines CTA button sizing.
   */
  size?: 'small' | 'medium'
  /**
   * Array of CTAs. Minimum of one, max of two.
   * @see CtaProps
   */
  ctas: MaxTwoCta
```

ref: https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMIBUCCyDeBYAKGWUgA8wAuZAZzClAHNCBfQwsATwAcUNNrkAXmQBtPgBo0WAPwBdNgQQB7ELWQIwcKnyG4SEclQDkMJUqPJWBQstVh1m6gAZtWAcJHzFKtRrjUARld+XRE-L1tfRwAmYPdRP0lwmx97P2oAZjjQxIc4JM1ZIA